sw/host/opentitantool/src/command/transport.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use erased_serde::Serialize;
+use std::any::Any;
+use structopt::StructOpt;
+
+use opentitanlib::app::command::CommandDispatch;
+use opentitanlib::app::TransportWrapper;
+
+/// Initialize state of a transport debugger device to fit the device under test.  This
+/// typically involves setting pins as input/output, open drain, etc. according to configuration
+/// files.
+#[derive(Debug, StructOpt)]
+pub struct TransportInit {}
+
+impl CommandDispatch for TransportInit {
+    fn run(
+        &self,
+        context: &dyn Any,
+        transport: &TransportWrapper,
+    ) -> Result<Option<Box<dyn Serialize>>> {
+        // Configure all GPIO pins to default direction and level, according to
+        // configuration files provided.
+        transport.apply_default_pin_configurations()?;
+        Ok(None)
+    }
+}
+
+/// Commands for interacting with the transport debugger device itself.
+#[derive(Debug, StructOpt, CommandDispatch)]
+pub enum InternalTransportCommand {
+    Init(TransportInit),
+}
+
+#[derive(Debug, StructOpt)]
+pub struct TransportCommand {
+    #[structopt(subcommand)]
+    command: InternalTransportCommand,
+}
+
+impl CommandDispatch for TransportCommand {
+    fn run(
+        &self,
+        _context: &dyn Any,
+        transport: &TransportWrapper,
+    ) -> Result<Option<Box<dyn Serialize>>> {
+        self.command.run(self, transport)
+    }
+}
```

Comment:
This seems like an extra layer of structure here.  Shouldn't `InternalTransportCommand` just be `TransportCommand`?

Reply:
I did not realize that I could do that.  Done now, it is definitely cleaner.